experiments/download_ssl4eo.py

```diff
@@ -42,120 +68,65 @@
 
 warnings.simplefilter("ignore", UserWarning)
 
-ALL_BANDS_L2A = [
-    "B1",
-    "B2",
-    "B3",
-    "B4",
-    "B5",
-    "B6",
-    "B7",
-    "B8",
-    "B8A",
-    "B9",
-    "B11",
-    "B12",
-]
-ALL_BANDS_L1C = [
-    "B1",
-    "B2",
-    "B3",
-    "B4",
-    "B5",
-    "B6",
-    "B7",
-    "B8",
-    "B8A",
-    "B9",
-    "B10",
-    "B11",
-    "B12",
-]
-RGB_BANDS = ["B4", "B3", "B2"]
-ALL_BANDS_GRD = ["VV", "VH"]
-
 
 def date2str(date: datetime) -> str:
     return date.strftime("%Y-%m-%d")
 
 
-def get_period(date: datetime, days: int = 5) -> Tuple[str, str]:
+def get_period(date: datetime, days: int = 5) -> Tuple[str, str, str, str]:
     date1 = date - timedelta(days=days / 2)
     date2 = date + timedelta(days=days / 2)
-    return date2str(date1), date2str(date2)
+    date3 = date1 - timedelta(days=365)
+    date4 = date2 - timedelta(days=365)
+    return (
+        date2str(date1),
+        date2str(date2),
+        date2str(date3),
+        date2str(date4),
+    )  # two-years buffer
```

Comment:
Nope, sampling whenever should also be fine

Reply:
> 

One potential limit is the max query frequency from GEE: https://developers.google.com/earth-engine/guides/usage#adjustable_quota_limits. I didn't meet this issue, but if using all >100 cores this may show up.